massa-execution-worker/src/execution.rs

```diff
@@ -445,6 +449,57 @@ impl ExecutionState {
             )));
         }
 
+        // Check selector
+        // Note 1: Has to be done after slot limit but before executed check
+        // Note 2: that this is done for a node to create a Block with 'fake' denunciation thus
+        //       include them in executed denunciation and prevent (by occupying the corresponding entry)
+        //       any further 'real' denunciation.
+
+        match &denunciation {
+            Denunciation::Endorsement(_de) => {
+                // Get selected address from selector and check
+                let selected = self.selector.get_selection(*de_slot);
+                match selected {
+                    Ok(selection) => {
+                        if let Some(address) = selection
+                            .endorsements
+                            .get(*denunciation.get_index().unwrap_or(&0) as usize)
+                        {
+                            if *address != addr_denounced {
+                                return Err(ExecutionError::IncludeDenunciationError(
+                                    "Attempt to execute a denunciation but address was not selected".to_string()
+                                ));
+                            }
+                        } else {
+                            // Panic here as this should never happen
+                            panic!("Could not get selected address for endorsements at index");
+                        }
+                    }
+                    Err(e) => {
+                        // We panic here as this should never happen
+                        panic!("Cannot get producer from selector: {}", e);
+                    }
+                }
+            }
+            Denunciation::BlockHeader(_de) => {
+                let selected_address = self.selector.get_producer(*de_slot);
+                match selected_address {
+                    Ok(address) => {
+                        if address != addr_denounced {
+                            return Err(ExecutionError::IncludeDenunciationError(
+                                "Attempt to execute a denunciation but address was not selected"
+                                    .to_string(),
+                            ));
+                        }
+                    }
+                    Err(e) => {
+                        // We panic here as this should never happen
+                        panic!("Cannot get producer from selector: {}", e);
+                    }
+                }
+            }
+        }
+
         // ignore the denunciation if it was already executed
         let de_idx = DenunciationIndex::from(denunciation);
         if context.is_denunciation_processed(&de_idx) {
```

Comment:
* this can be moved before the PoS check
* processed => executed

Reply:
Done

Reply:
Will be done il a followup